### PR TITLE
Make FS use domains.

### PR DESCRIPTION
**Pull request by SciresM**
Adds new serviceIpcPrepareHeader, serviceIpcParse, serviceCreateSubservice functions.

Causes a breaking (but necessary) change to ipcParseForDomain.

Makes fsInitialize() call serviceConvertToDomain(), resulting in all FS commands using domain object IDs.

### CODE REVIEW

**Comment by SciresM** [comment]
Ideally over time we can refactor other services to use these primitives. 

QuestionL Should we eventually break ipcParse() and make it take a size? It would be nice to have the RawSize be more meaningful in IpcParsedCommand...

**Comment by SciresM** [comment]
Oops, just realized this actually breaks IpcParsedCommand support for domain on the serverside. Should be straightforward, will push a fix tomorrow morning.

File: nx/include/switch/services/sm.h [hunk 1]
```diff
@@ -113,6 +113,20 @@ static inline void serviceCreateDomainSubservice(Service* s, Service* parent, u3
     s->object_id = object_id;
 }
 
+/**
+ * @brief Creates a domain subservice object from a parent service.
```
Comment (yellows8) [inline nx/include/switch/services/sm.h]: Update these comments, it supports regular serviceCreate too.

File: nx/include/switch/services/sm.h [hunk 2]
```diff
@@ -113,6 +113,20 @@ static inline void serviceCreateDomainSubservice(Service* s, Service* parent, u3
...
+ * @param[out] s Service object.
+ * @param[in] parent Parent service, necessarily a domain or domain subservice.
+ * @param[in] object_id Object ID for this subservice.
```
Comment (yellows8) [inline nx/include/switch/services/sm.h]: This param doesn't exist & the r/i params are missing in these comments.

Comment (yellows8) [inline nx/include/switch/services/sm.h]: This param doesn't exist & the r/i params are missing in these comments.

**Pull request merged**